promtimer/cbstats.py

```diff
@@ -802,6 +823,57 @@ def get_prometheus_times(cbcollect_dir):
     return min(min_times), max(max_times)
 
 
+def parse_couchbase_timezone(cbcollect_dir):
+    """
+    Parses the timezone output from cat /etc/timezone from couchbase.log
+    """
+    log_file = path.join(cbcollect_dir, 'couchbase.log')
+    with open(log_file, 'r', errors='replace') as file:
+        it = iter(file)
+        while (line := next(it, None)) is not None:
+            if line == 'cat /etc/timezone\n':
+                separator = next(it, None)
+                assert separator.startswith(
+                    '=' * 70), 'Expected separator, not {}'.format(separator)
+                timezone = next(it, None)
+                if timezone is None or ' ' in timezone:
+                    raise ValueError('Missing /etc/timezone output')
+
+                timezone = timezone.strip()
+                # Observed just /UTC in some environments
+                if timezone.startswith('/'):
+                    return 'Etc' + timezone
+                return timezone
+
+
+def parse_cbcollect_info_timezone(cbcollect_dir):
```

Comment:
Since we don't always extract cbcollect_info, this function misses it generally unless the user explicitly extracts this file -- or the whole zipped archive. Personally I almost never extract the whole archive any more as it gobbles up so much disk space. (I just run promtimer and then make liberal use of `unzip -C`).

In any case, it would be nice if this function would work with something extracted from the zip file (if present) or else look for the file on disk. See what I did with `get_my_user_log` and `parse_user_log`. Overall it's not a huge deal, but it would be nice.

Reply:
Done